app/models/manageiq/providers/amazon/agent_coordinator_worker/runner/response_thread.rb

```diff
@@ -0,0 +1,190 @@
+module ManageIQ::Providers::Amazon::AgentCoordinatorWorker::Runner::ResponseThread
+  require 'xml/xml_utils'
+  require 'scanning_mixin'
+  include ScanningMixin
+
+  #
+  # This thread startup method is called by the AgentCoordinatorWorker::Runner
+  # at the end of the do_before_work_loop method.
```

Comment:
Actually no.  That's one reason it might be better to implement this as a worker.  If the AgentCoordinatorWorker dies then it will be restarted and the thread will too.  But if the worker lives and the thread dies then no.  I can look into adding something if you think that is something we should do, or compare and contrast that effort with that of making it a separate worker.

Reply:
Keep it as is for now. We'll address it at a later time.

Reply:
We must track this thread and handle it dying within the worker context.